### ISSUE TITLE
New Rule: No top level await

### ISSUE THREAD

**Issue by ArnaudBarre** [eslint/eslint#15339]
### Rule details

Disallow top level await

### Related ECMAScript feature

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await

### What type of rule is this?

Warns about a potential problem

### Example code

```js
const value = await someBrowserAsyncApi()
```

### Why should this rule be in the core instead of a plugin?

I'm not sure about that. It could also be proposed as a negation of eslint-plugin-unicorn/prefer-top-level-await, but not sure this would feat in the philosophy of the package.

### Participation

- [X] I am willing to submit a pull request to implement this rule.

### Additional comments

While this a great addition for node or full ESM code, it can result to unexpected behaviour in modern JS code once bundled.

**Comment by nzakas** [comment]
Thanks for the suggestion.  We aren’t adding any more rules that just disallow syntax. You can use no-restricted-syntax to prevent any syntax that you’d like.
